app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/attributeStrategies.ts

```diff
@@ -1,82 +1,156 @@
 // In an OOP Language -
+
+import { FeatureAttributeName } from "../states/map/FeatureAttributeName";
+
 // TypeScript
-interface FeatureAttributeFunctions {
+interface FeatureAttributeTypes {
     isNumeric() : boolean
     isString(): boolean
+    getFilterType(): string
+    ignore(): boolean
+    attributeName: FeatureAttributeName
 }
 
-enum Attributes {
-    LABEL = "label",
-    TYPE = "type",
-    URI = "uri",
-    FREQUENCY = "frequency",
-    TIMESTAMP = "timestamp",
-    LOCALITY = "locality",
-    POSTALCODE = "postalcode",
-    REGIONS =  "regions",
-    TRANSMISSIONPOWER = "transmissionPower",
-    GEOMETRY = "geometry",
-}
 
-export class FeatureAttribute implements FeatureAttributeFunctions{
-    private attributeName: string 
-    private subclass: any
-    constructor(attributeName: string) {
+ export abstract class FeatureAttribute implements FeatureAttributeTypes{
+    public attributeName: FeatureAttributeName 
+    constructor(attributeName: FeatureAttributeName) {
         this.attributeName = attributeName
-        this.subclass = null
-        switch (this.attributeName) {
-            case (Attributes.TIMESTAMP): {
-                this.subclass = FeatureAttributeTimestamp.getInstance()
+    }
+    static AttributeTypeOf(attributeName: FeatureAttributeName): FeatureAttribute {
+        switch(attributeName) {
+            case FeatureAttributeName.timestamp: {
+               return TimestampFeatureAttribute.getInstance(attributeName)
             }
-        }
+            case FeatureAttributeName.frequency: {
+                return FrequencyFeatureAttribute.getInstance(attributeName)
+            }
+            case FeatureAttributeName.transmissionPower: {
+                return TransmissionPowerFeatureAttribute.getInstance(attributeName)
+            }
+            
+            default : {
+                return IgnoreFeatureAttribute.getInstance(attributeName)
+                
+            }
+         }
     }
 
-    isNumeric() : boolean {
-        return (this.subclass? this.subclass.isNumeric() : false)
+    ignore(): boolean {
+        return false
+    }
+    getFilterType(){
+        return 'none'
     }
 
-    isString() : boolean {
-        return (this.subclass? this.subclass.isString() : false)
+    abstract isNumeric() : boolean 
 
-    }
-    
+    abstract isString() : boolean 
     
+}
 
+class TimestampFeatureAttribute extends FeatureAttribute implements FeatureAttributeTypes {
+    private static instance: TimestampFeatureAttribute;
```

Comment:
Public instance:
static readonly instance: TimestampFeatureAttribute = new TimestampFeatureAttribute();

Private constructor. Constructor doesn't need the attribute name, since it's always the same. Make it an abstract method implementation as I suggested above.